web/channels/player_channel.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Yggdrasil.PlayerChannel do
+  use Yggdrasil.Web, :channel
+  alias Yggdrasil.Player
+  alias Yggdrasil.Endpoint
+
+  def join("player:" <> user_id = topic, _message, socket) do
+    if socket.assigns.user == String.to_integer(user_id) do
+      socket = assign socket, :player_topic, topic
+      {:ok, socket}
+    else
+      {:error, %{ error: "auth failure"} }
+    end
+  end
+
+  def handle_in("join_game", _message, socket) do
+    player_topic = socket.assigns.player_topic
+    push_msg = fn (payload) ->
+      Endpoint.broadcast! player_topic, "event", payload
+    end
+
+    case Player.join_game socket.assigns.user, self, push_msg do
```

Comment:
I do the player join here so that the system can immediately push the welcome message back to the channel. If I did that in the channel's join then the user would never see the message. I could duplicate the channel's protocol with the player process, where you first have to create it, then tell it to join.

I think I'm going to end up using the "join" to just register the player with the system. The next step would be to make the user say which character to join as, which game to join potentially, or if they want to create a new character.